card/db/migrate/20150220141031_create_deck_routes_file.rb

```diff
@@ -0,0 +1,14 @@
+# -*- encoding : utf-8 -*-
+
+class CreateDeckRoutesFile < ActiveRecord::Migration
+  def up
+    routes_path = File.join Wagn.root, 'config', 'routes.rb'
+    if !File.exist? routes_path
+      template_path = File.join Wagn.gem_root, 'lib', 'wagn', 'generators', 'wagn', 'templates', 'config', 'routes.erb'
+      @include_jasmine_engine = false
+      decko_path = ''
+      routes = ERB.new(File.read template_path).result binding
+      File.open(routes_path, "w") { |f|  f.write routes }
+    end
+  end
+end
```

Comment:
I may be finally getting something here.

My understanding:
We will have wagn/lib/decko/engine.rb, which is more or less the file from the decko-rails gem.  That is what is mounted in an engine app, and what is mounted in wagn/.../config/routes.rb.

Are you saying there will be another class MyAppEngine < Decko::Engine ?  I'm not sure we would need that.  My original thinking was that Decko::Engine.config would be connected by railties with MyApp.config, but now I'm not sure if that is desirable or possible.  Next thought is that we can copy the ones Decko needs from the application.config object in one of the initializers.

Do you have any better understanding and suggestions about how this should work?